### DISCUSSION TITLE
Use of fixed time step with PeleC

### DISCUSSION THREAD

**Discussion by JEDurand** [AMReX-Combustion/PeleC#863] [Q&A] [answered]
Dear community,

I am writing to you because I would like to obtain a fixed time step during my simulations to sample regularly my snapshots. However, when using "fixed_dt", I don't get what I am looking for. I obtained another time step than I wrote in my input file and these time steps change during the simulation. 

How can I use the options offered by PeleC to set the time step and its value during the simulation ?

Thank you, in advance, for your help.

**Comment by marchdf** [comment, marked as answer]
I am curious if the other user controls are getting in the way. I would recommend using:
```
pelec.init_shrink    = 1.0 
pelec.change_max     = 1.0
pelec.fixed_dt = 1e-8
```

The first time step taken by PeleC is going to be affected by the `init_shrink` factor. This is to allow a slow ramp up of the time step. Does trying that fix your problem?

**Reply by marchdf** [reply]
See also answer below by @baperry2  for an alternate way.

**Comment by baperry2** [comment]
You can also use `amr.plot_per = 1.0e-4` to dump plot files at fixed time periods rather than numbers of time steps if that is the reason you are trying to used fixed time step sizes. Note, with this command the plot files will be dumped at the closest timestep to the desired time interval - this is good enough in many cases, but if you need data at an exact time running with fixed_dt is the way to go.

**Comment by JEDurand** [comment]
Thank you very much for your help and answers. I tried it,  the time step is now set with the expected value. 
Of course, I would use a regular sampling. However, for academic purpose, I need a regular time step.
